internal/manifests/manifests.go

```diff
@@ -0,0 +1,219 @@
+
+var _ restapi.ManifestsAPI = &Manifests{}
+
+const manifestFolder = "manifests"
+
+// NewManifestsAPI returns manifests API
+func NewManifestsAPI(db *gorm.DB, log logrus.FieldLogger, objectHandler s3wrapper.API) *Manifests {
+	return &Manifests{
+		db:            db,
+		log:           log,
+		objectHandler: objectHandler,
+	}
+}
+
+// Manifests represents manifests handler implementation
+type Manifests struct {
+	db            *gorm.DB
+	log           logrus.FieldLogger
+	objectHandler s3wrapper.API
+}
+
+func (m *Manifests) CreateClusterManifest(ctx context.Context, params operations.CreateClusterManifestParams) middleware.Responder {
+	log := logutil.FromContext(ctx, m.log)
+	log.Infof("Creating manifest in cluster %s", params.ClusterID)
+
+	if params.CreateManifestParams.Folder == nil {
+		defaultFolder := models.CreateManifestParamsFolderManifests
+		params.CreateManifestParams.Folder = &defaultFolder
+	}
+
+	cluster, apierr := GetCluster(ctx, m.log, m.db, params.ClusterID.String())
+	if apierr != nil {
+		return common.GenerateErrorResponder(apierr)
+	}
+
+	fileName := filepath.Join(*params.CreateManifestParams.Folder, *params.CreateManifestParams.FileName)
+	manifestContent, err := base64.StdEncoding.DecodeString(*params.CreateManifestParams.Content)
+	if err != nil {
+		log.Errorf("Cluster manifest %s for cluster %s failed to base64 decode: [%s]",
+			fileName, cluster.ID, *params.CreateManifestParams.Content)
+		return common.GenerateErrorResponderWithDefault(errors.New("failed to base64-decode cluster manifest content"), http.StatusBadRequest)
+	}
+
+	objectName := GetManifestObjectName(*cluster.ID, fileName)
+	if err := m.objectHandler.Upload(ctx, manifestContent, objectName); err != nil {
+		return common.GenerateErrorResponder(fmt.Errorf("failed to upload %s to s3", objectName))
+	}
+
+	log.Infof("Done creating manifest %s for cluster %s", fileName, cluster.ID)
+	manifest := models.Manifest{FileName: *params.CreateManifestParams.FileName, Folder: *params.CreateManifestParams.Folder}
+	return operations.NewCreateClusterManifestCreated().WithPayload(&manifest)
+}
+
+func (m *Manifests) ListClusterManifests(ctx context.Context, params operations.ListClusterManifestsParams) middleware.Responder {
+	log := logutil.FromContext(ctx, m.log)
+	log.Infof("Listing manifests in cluster %s", params.ClusterID)
+
+	cluster, apierr := GetCluster(ctx, m.log, m.db, params.ClusterID.String())
+	if apierr != nil {
+		return apierr
+	}
+
+	objectName := filepath.Join(cluster.ID.String(), manifestFolder)
+	files, err := m.objectHandler.ListObjectsByPrefix(ctx, objectName)
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+
+	manifests := models.ListManifests{}
+	for _, file := range files {
+		parts := strings.Split(file, "/")
+		if len(parts) > 2 {
+			manifests = append(manifests, &models.Manifest{FileName: filepath.Join(parts[3:]...), Folder: parts[2]})
+		} else {
+			log.Warnf("Cannot list file %s in cluster %s", file, cluster.ID)
```

Comment:
Do we want to return partial list? it looks like it can hide some bugs no?

Reply:
If it should never happen better to log and return error

Reply:
Even if it will be a log i think that no one will ever going to see it